docs/content/ingestion/update-existing-data.md

```diff
@@ -1,46 +1,57 @@
+the ingested data. There are several ways this can be done.
 
-##### Reindexing
-You ingested some raw data to a dataSource A and later you want to re-index and create another dataSource B which has a subset of columns or different granularity. Or, you may want to change granularity of data in A itself for some interval.
+##### Updating Dimension Values
 
-##### Delta-ingestion
-You ingested some raw data to a dataSource A in an interval, later you want to "append" more data to same interval. This might happen because you used realtime ingestion originally and then received some late events. 
+If you have a dimension where values need to be updated frequently, try first using [lookups](../querying/lookups.html). A 
+classic use case of lookups is when you have an ID dimension stored in a Druid segment, and want to map the ID dimension to a 
+human-readable String value that may need to be updated periodically.
 
-Here are the Druid Features you could use to achieve above updates.
+##### Rebuilding Segments (Reindexing)
 
-- You can use batch ingestion to override an interval completely by doing the ingestion again with the raw data.
-- You can use re-indexing and delta-ingestion features provided by batch ingestion.
+If lookups are not sufficient, Druid segments may need to be rebuilt as segments are immutable once created. Rebuilding a segment 
+is known as reindexing the data. For example, if you want to add or remove columns from your existing segments, or you want to 
+change the rollup granularity of your segments, you will have to reindex your data.
 
+We recommend keeping a copy of your raw data around in case you ever need to reindex your data.
 
-### Re-indexing and Delta Ingestion with Hadoop Batch Ingestion
+##### Dealing with Delayed Events (Delta Ingestion)
 
-This section assumes the reader has understanding of batch ingestion using Hadoop. See [HadoopIndexTask](../misc/tasks.html#index-hadoop-task) and further explained in [batch-ingestion](batch-ingestion.md). You can use hadoop batch-ingestion to do re-indexing and delta-ingestion as well.
+If you have a batch ingestion pipeline and have delayed events come in and want to append these events to existing 
+segments and avoid the overhead of rebuilding new segments with reindexing, you can use delta ingestion.
 
-It is enabled by how Druid reads input data for doing hadoop batch ingestion. Druid uses specified `inputSpec` to know where the data to be ingested is located and how to read it. For simple hadoop batch ingestion you would use `static` or `granularity` spec types  which allow you to read data stored on HDFS.
+### Reindexing and Delta Ingestion with Hadoop Batch Ingestion
 
-There are two other `inputSpec` types to enable reindexing and delta-ingestion.
+This section assumes the reader has understanding of batch ingestion using Hadoop. See [HadoopIndexTask](../misc/tasks.html#index-hadoop-task) 
```

Comment:
instead of "has understanding of", "has an understanding of" or just "understands"